tools/run_cargo_test.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Utility script to invoke cargo test within the current Python environment.
+
+Notably, this sets up the environment variables necessary for Qiskit to be
+found by PyO3 / our Rust test executable.
+"""
+
+import os
+import subprocess
+import sys
+import sysconfig
+
+# This allows the Python interpreter baked into our test executable to find the
+# Qiskit installed in the active environment.
+os.environ["PYTHONUSERBASE"] = sys.prefix
+
+# On Linux, the test executable's RPATH doesn't contain libpython, so we add it
+# to the dlopen search path here.
+os.environ["LD_LIBRARY_PATH"] = os.pathsep.join(
+    filter(None, [sysconfig.get_config_var("LIBDIR"), os.getenv("LD_LIBRARY_PATH")])
+)
```

Comment:
Is there a reason to not just do?:
```suggestion
os.environ["LD_LIBRARY_PATH"] = os.pathsep.join([
    sysconfig.get_config_var("LIBDIR"),
    os.getenv("LD_LIBRARY_PATH")
])
```

I've always found the `filter(None, ...)` pattern a bit odd when you can just use a generator expression.

Reply:
The reason I did that was to filter out `None` in the case that `"LD_LIBRARY_PATH"` isn't actually set to anything. I can do something different if you prefer!